lms/djangoapps/bulk_email/tasks.py

```diff
@@ -496,22 +515,37 @@ def _send_course_email(entry_id, email_id, to_list, global_email_context, subtas
 
             except SMTPDataError as exc:
                 # According to SMTP spec, we'll retry error codes in the 4xx range.  5xx range indicates hard failure.
+                total_recipients_failed += 1
+                log.warning(
+                    "Status: Failed(SMTPDataError), Email not sent to recipient no: %s, Email address: %s",
+                    recipient_no,
+                    email
+                )
                 if exc.smtp_code >= 400 and exc.smtp_code < 500:
                     # This will cause the outer handler to catch the exception and retry the entire task.
                     raise exc
                 else:
                     # This will fall through and not retry the message.
-                    log.warning('Task %s: email with id %s not delivered to %s due to error %s', task_id, email_id, email, exc.smtp_error)
+                    log.warning('Task %s: email with id %s not delivered to %s due to error %s',
+                                task_id, email_id, email, exc.smtp_error)
                     dog_stats_api.increment('course_email.error', tags=[_statsd_tag(course_title)])
                     subtask_status.increment(failed=1)
 
             except SINGLE_EMAIL_FAILURE_ERRORS as exc:
                 # This will fall through and not retry the message.
+                total_recipients_failed += 1
+                log.warning(
+                    "Status: Failed(SINGLE_EMAIL_FAILURE_ERRORS), Recipient no: %s, Email address: %s",
+                    recipient_no,
+                    email
+                )
                 log.warning('Task %s: email with id %s not delivered to %s due to error %s', task_id, email_id, email, exc)
                 dog_stats_api.increment('course_email.error', tags=[_statsd_tag(course_title)])
                 subtask_status.increment(failed=1)
```

Comment:
We are already counting failures here in the subtask_status -- do we also need to count it with total_recipients_failed?